src/python/pants/backend/python/target_types_rules.py

```diff
@@ -209,7 +239,7 @@ async def resolve_python_distribution_entry_points(
     # Check that we only have targets with a pex entry_point field.
     for target in targets:
         if not target.has_field(PexEntryPointField):
-            raise InvalidFieldException(
+            raise InvalidEntryPoint(
```

Comment:
This got my head scratching. I'd like to be consistent, but some code raise InvalidFieldException, where as other InvalidEntryPoint... feels like InvalidEntryPoint ought to inherit from InvalidFieldException too, at least... so a catch for field exceptions would catch this.

Any way, I went with keeping the tests as is, as that is the user facing interface. But I think we could make a do-over here to make it more consistent. ;)

Reply:
That sounds good to have `InvalidEntryPoint` subclass `InvalidFieldException`, definitely. And makes sense if you want to do in a followup to keep this PR better scoped.

Reply:
Issue #12430